docs: Add static installation guide

jodh-intel opened the pull request:
Add a document explaining how to (and why you would want to) install the
static version of Kata Containers.

Fixes: #470.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

jodh-intel commented:
Although https://github.com/kata-containers/packaging/pull/654 has now landed, we can't merge this until we have a new release containing the script, so adding dnm...

jodh-intel commented:
Related: https://github.com/kata-containers/packaging/issues/655.

jodh-intel commented:
@klynnrif - ptal.

jodh-intel commented:
Blocked on https://github.com/kata-containers/tests/pull/1869.

jodh-intel commented:
Now blocked on a new static binaries release archive containing the script.

klynnrif commented on install/installing-kata-static-binaries.md:
```diff
@@ -0,0 +1,47 @@
+>
+> This installation method is **not** recommended for normal users since the
+> installation does not use the distribution package manager. This means new
+> versions of Kata Containers will **not** be automatically installed.
```
Line 20 suggested rewrite (for an active voice): 
versions of Kata Containers will **not** automatically install.

jodh-intel replied:
Thanks @klynnrif - branch updated.

jodh-intel commented:
This is now ready for re-review.

/test

jodh-intel commented:
The CI found a bug (yay), so this is now blocked on https://github.com/kata-containers/packaging/pull/801 making its way into a release.

jodh-intel commented:
@jcvenegas, @egernst - have we changed the way we produce static binaries recently [1]? fwics we used to use https://github.com/kata-containers/packaging/blob/master/release/kata-deploy-binaries.sh, but now either we don't use this script or something is post-processing the static archive to remove [the script I added](https://github.com/kata-containers/packaging/blob/master/static-build/scripts/kata-configure-docker.sh).

---
[1] - 1.9.1 and 1.9.2 contain the expected file, but newer releases don't :cry:

jcvenegas commented:
@jodh-intel It is partially used - It is sourced instead of being used as a cli script with the main idea of do a parallel job, so sounds like a bug in https://github.com/kata-containers/kata-containers/blob/master/.github/workflows/main.yaml

grahamwhaley commented:
@jodh-intel - is this still alive - want me to review/nudge it, or does it need killing?

jodh-intel commented:
It fell off the stack I'm afraid. I think we should keep it open since if we can land this, we get "free" testing of the static binaries.

If anyone wants to volunteer to pick this up, just shout! ;) Otherwise, I'll try to move it up my backlog a bit... ;)

jodh-intel commented:
Closing this since time has moved on:

- Cloud users should be using [`kata-deploy` (1.x)](https://github.com/kata-containers/packaging/tree/master/kata-deploy) (or the [2.0 version](https://github.com/kata-containers/kata-containers/tree/2.0-dev/tools/packaging/kata-deploy)).
- [Distro packaging is effectively "done"](https://github.com/kata-containers/packaging/wiki/Distro-packaging-status) - we'd rather users use those versions.
- [We already have static instructions for 2.0](https://github.com/kata-containers/kata-containers/pull/739).
